diff --git a/wan/__init__.py b/wan/__init__.py
index c49c77e..5ac217c 100644
--- a/wan/__init__.py
+++ b/wan/__init__.py
@@ -1,7 +1,3 @@
 # Copyright 2024-2025 The Alibaba Wan Team Authors. All rights reserved.
 from . import configs, distributed, modules
-from .image2video import WanI2V
-from .speech2video import WanS2V
-from .text2video import WanT2V
 from .textimage2video import WanTI2V
-from .animate import WanAnimate
\ No newline at end of file
